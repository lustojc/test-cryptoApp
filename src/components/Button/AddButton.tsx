interface buttonProps {
  onClickAdd: (id: number, name: string, price: number) => void;
  rank: string;
  price: number;
  text: string;
  name: string;
}

export default function AddButton({ onClickAdd, rank, name, price, text }: buttonProps) {
  return (
    <div>
      <button className="crypto-block__btn" onClick={() => onClickAdd(+rank, name, price)}>
        {text}
      </button>
    </div>
  );
}
