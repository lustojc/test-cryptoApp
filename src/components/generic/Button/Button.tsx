import { ButtonHTMLAttributes } from 'react';

interface props extends ButtonHTMLAttributes<HTMLButtonElement> {
  text: string;
  backgroundColor: string;
  color: string;
  size: 'lg' | 'md' | 'sm';
  dataAtt: string;
  borderRadius?: string;
  type?: 'submit' | 'reset' | 'button';
  onClickBtn?: () => void;
}

export default function Button({
  text = 'Button',
  backgroundColor = 'olive',
  color = 'white',
  size = 'md',
  dataAtt,
  onClickBtn,
  borderRadius,
  type,
}: props) {
  return (
    <button
      onClick={onClickBtn}
      data-cy={dataAtt}
      type={type}
      className={`button button-backgroundColor__${backgroundColor} button-size__${size} button-color__${color} button-borderRadius__${borderRadius} `}>
      {text}
    </button>
  );
}
