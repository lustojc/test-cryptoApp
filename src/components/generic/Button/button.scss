.button {
  cursor: pointer;
  border: none;

  &-backgroundColor {
    &__olive {
      background-color: rgb(128, 130, 8);
    }

    &__darkGrey {
      background-color: rgba(0, 0, 0, 0.637);
    }
    &__white {
      background-color: white;
    }
  }
  &-color {
    &__white {
      color: white;
    }
    &__black {
      color: black;
    }
  }
  &-size {
    &__lg {
      padding: 15px 20px;

      @media (max-width: $breakpoint-tablet-large) {
        padding: 10px 15px;
      }
    }
    &__md {
      padding: 10px 15px;

      @media (max-width: $breakpoint-tablet-large) {
        padding: 5px 6.5px;
      }
    }
    &__sm {
      padding: 5px 7px;

      @media (max-width: $breakpoint-tablet-large) {
        padding: 10px 15px;
      }
    }
  }

  &-borderRadius {
    &__rounded {
      border-radius: 10px;
    }
  }
}
