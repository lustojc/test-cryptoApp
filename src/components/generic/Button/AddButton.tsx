import { useEffect, useState } from 'react';

import { useAppDispatch } from '../../../libs/hooks/hooks';
import { useScrollBlock } from '../../../libs/hooks/useScrollBlock';

import { addItems } from '../../../store/slices/portfolioSlice';

import InputForm from '../../Modal/ModalInput';

import Button from './Button';

interface buttonProps {
  rank: string;
  price: number;
  text: string;
  id: string;
  name: string;
}

export default function AddButton({ rank, id, price, text, name }: buttonProps) {
  const [formActive, setFormActive] = useState<boolean>(false);
  const [blockScroll, allowScroll] = useScrollBlock();

  const dispatch = useAppDispatch();

  const onClickAdd = (value: string) => {
    if (+value > 0) {
      const item = {
        id: rank,
        title: id,
        price: price,
        amount: value,
        name: name,
      };
      dispatch(addItems(item));
      setFormActive(!formActive);
    } else {
      alert('Incorrect value');
    }
  };

  useEffect(() => {
    if (formActive) {
      blockScroll();
    }
    return () => {
      allowScroll();
    };
  }, [formActive]);

  useEffect(() => {
    const closeModal = (e: { keyCode: number }) => {
      if (e.keyCode === 27) {
        setFormActive(false);
      }
    };
    window.addEventListener('keydown', closeModal);
    return () => window.removeEventListener('keydown', closeModal);
  }, []);

  const onClickBtn = () => {
    setFormActive(!formActive);
  };

  return (
    <div>
      <Button
        text={text}
        backgroundColor={'olive'}
        color={'white'}
        size={'md'}
        onClickBtn={onClickBtn}
        dataAtt="addBtn"
      />
      {formActive && <InputForm onClickAdd={onClickAdd} setFormActive={setFormActive} />}
    </div>
  );
}
