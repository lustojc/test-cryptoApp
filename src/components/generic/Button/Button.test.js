import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import Button from './Button';

describe('test button component', () => {
  it('render button', () => {
    render(<Button text="Add" color="white" />);

    const btnElement = screen.getByText(/add/i);
    expect(btnElement).toBeInTheDocument();
    expect(btnElement).toHaveClass('button-color__white');
    expect(btnElement).toMatchSnapshot();
  });
});
