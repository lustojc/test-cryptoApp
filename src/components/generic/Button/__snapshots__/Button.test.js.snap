// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test button component render button 1`] = `
<button
  class="button button-backgroundColor__olive button-size__md button-color__white button-borderRadius__undefined "
>
  Add
</button>
`;
